services/121-service/src/notifications/lookup/lookup.service.ts

```diff
@@ -20,14 +20,32 @@ export class LookupService {
     return { result: numberCorrect };
   }
 
-  public async lookupAndCorrect(phoneNumber: string): Promise<string> {
+  public async lookupAndCorrect(
+    phoneNumber: string,
+    throwNoException?: boolean,
+  ): Promise<string> {
     try {
+      // Add additional sanitizing (incl NL-specific) because user is given no opportunity to correct here
+      const updatedPhone =
+        phoneNumber.substr(0, 2) == '00'
+          ? phoneNumber.substr(2, 20)
+          : phoneNumber.substr(0, 2) == '06'
+          ? '31' + phoneNumber
+          : phoneNumber.substr(0, 3) == '+00'
+          ? phoneNumber.substr(3, 20)
+          : phoneNumber.substr(0, 2) == '+0'
+          ? phoneNumber.substr(2, 20)
+          : phoneNumber;
+
       const lookupResponse = await twilioClient.lookups
-        .phoneNumbers(phoneNumber)
+        .phoneNumbers(updatedPhone)
         .fetch({ type: ['carrier'] });
       return lookupResponse.phoneNumber.replace(/\D/g, '');
     } catch (e) {
       console.log('e: ', e);
+      if (throwNoException) {
```

Comment:
In this particular use-case of this function, it should not throw an exception. This way the whole csv-file can be checked and the nr of invalid phone-numbers counted. But at other places, where this function is used, you do want an exception to be thrown, because the request involves only 1 phone-number, instead of a list of phone-numbers. That was the idea at least.. I realize this can probably be done better.